stdlib/ostruct.rb

```diff
@@ -60,10 +66,52 @@ def to_n
   def hash
     @table.hash
   end
+  
+  def table
+    @table
+  end
+  
+  def delete_field(name)
+    sym = name.to_sym
+    begin
+      singleton_class.__send__(:remove_method, sym, "#{sym}=")
+    rescue NameError
+    end
+    @table.delete sym
+  end
+  
+  def new_ostruct_member(name)
```

Comment:
Yep. table is a protected method in the MRI impl since ostensibly they want people to be able to subclass openstruct.

The rubyspecs for new_ostruct... also expect to be able to set the `@table` instance variable, so are you OK with not using `self.$$table` ?